{"id": 41478, "wybug_id": "wooyun-2015-088677", "wybug_title": "Google键盘未知客户导致的拒绝服务", "wybug_corp": "GOOGLE", "wybug_author": "elong", "wybug_date": "2015-01-04 10:14", "wybug_open_date": "2015-04-04 10:16", "wybug_type": "拒绝服务", "wybug_level": "低", "wybug_rank_0": "2", "wybug_status": "未联系到厂商或者厂商积极忽略", "wybug_from": "http://www.wooyun.org", "wybug_tags": ["拒绝服务"], "wybug_detail": "漏洞详情 披露状态：   \t\t\t\t\t\t\t\t\t2015-01-04：\t积极联系厂商并且等待厂商认领中，细节不对外公开\t\t\t\t\t\t\t\t\t2015-04-04：\t厂商已经主动忽略漏洞，细节向公众公开  简要描述： Google键盘在特定场景之下，会被拒绝服务攻击，导致用户弃用Google键盘 详细说明：  发生问题的调用栈如下：\n12-25 13:29:33.730: E/AndroidRuntime(12478): FATAL EXCEPTION: main12-25 13:29:33.730: E/AndroidRuntime(12478): Process: com.google.android.inputmethod.latin, PID: 1247812-25 13:29:33.730: E/AndroidRuntime(12478): java.lang.RuntimeException: Unable to resume activity {com.google.android.inputmethod.latin/com.android.inputmethod.dictionarypack.DictionarySettingsActivity}: java.lang.IllegalArgumentException: the bind value at index 1 is null12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:3007)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3036)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2334)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.ActivityThread.access$1000(ActivityThread.java:143)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.ActivityThread$H.handleMessage(ActivityThread.java:1242)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.os.Handler.dispatchMessage(Handler.java:102)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.os.Looper.loop(Looper.java:136)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.ActivityThread.main(ActivityThread.java:5289)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat java.lang.reflect.Method.invokeNative(Native Method)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat java.lang.reflect.Method.invoke(Method.java:515)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:849)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:665)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat dalvik.system.NativeStart.main(Native Method)12-25 13:29:33.730: E/AndroidRuntime(12478): Caused by: java.lang.IllegalArgumentException: the bind value at index 1 is null12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.database.sqlite.SQLiteProgram.bindString(SQLiteProgram.java:164)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.database.sqlite.SQLiteProgram.bindAllArgsAsStrings(SQLiteProgram.java:200)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:47)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1332)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.database.sqlite.SQLiteDatabase.queryWithFactory(SQLiteDatabase.java:1176)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1047)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1253)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat com.android.inputmethod.dictionarypack.MetadataDbHelper.getMetadataUriAsString(MetadataDbHelper.java:330)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat com.android.inputmethod.dictionarypack.MetadataDbHelper.isClientKnown(MetadataDbHelper.java:315)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat com.android.inputmethod.dictionarypack.DictionarySettingsFragment.onResume(DictionarySettingsFragment.java:126)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.Fragment.performResume(Fragment.java:1756)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.FragmentManagerImpl.moveToState(FragmentManager.java:931)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1071)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1053)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.FragmentManagerImpl.dispatchResume(FragmentManager.java:1872)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.Activity.performResume(Activity.java:5394)12-25 13:29:33.730: E/AndroidRuntime(12478): \tat android.app.ActivityThread.performResumeActivity(ActivityThread.java:2984)12-25 13:29:33.730: E/AndroidRuntime(12478): \t... 12 more\n问题触发场景this.mClientId = null 导致了这个问题。代码试图对异常情况进行处理并记录LOG，但不幸的是，异常发生在这段代码之前。\n\n   漏洞证明：  通过以下命令，可以重现问题：\nadb shell am start -n com.google.android.inputmethod.latin/com.android.inputmethod.dictionarypack.DictionarySettingsActivity\n\n\n   修复方案：  N/A   版权声明：转载请注明来源 elong@乌云\n ", "wybug_reply": "漏洞回应  厂商回应： 未能联系到厂商或者厂商积极拒绝  ", "replys": "漏洞评价：\n评论\n  \n\n\n", "wybug_level_fromcorp": "无影响厂商忽略", "wybug_rank_fromcorp": 0, "Ranks": null}